motr/io_req.c

```diff
@@ -1615,27 +1612,10 @@ static int ioreq_dgmode_read(struct m0_op_io *ioo, bool rmw)
 
 	m0_htable_for(tioreqht, ti, &xfer->nxr_tioreqs_hash) {
 		/*
-		 * Data was retrieved successfully, so no need to check the
-		 * state of the device.
+		 * Data was retrieved successfully from this target.
 		 */
 		if (ti->ti_rc == 0)
 			continue;
-
-		/* state is already queried in device_check() and stored
-		 * in ti->ti_state. Why do we do this again?
-		 */
-		rc = m0_poolmach_device_state(
-			pm, ti->ti_obj, &state);
-		if (rc != 0)
-			return M0_ERR(rc);
-		M0_LOG(M0_INFO, "device state for "FID_F" is %d",
-		       FID_P(&ti->ti_fid), state);
-		ti->ti_state = state;
-
-		if (!M0_IN(state, (M0_PNDS_FAILED, M0_PNDS_OFFLINE,
```

Comment:
In the old implementation if the device is in one of these states, the Motr client will not trigger Degraded Read, right?

So, removing such checking will enable degraded read if there is an error regardless the state of the device?

Reply:
No and yes. No - If the device is in one of these states, Motr client will trigger the Degraded Read. Yes - removing such checking will enable degraded read if there is an error, regardless of the device state.

Reply:
Understood.

This is a change which affects the behavior of Motr client in case of I/O failure and its handling.
As you explained in the commit message, original the Motr client only switches to Degraded Mode when Hare declares some device is in one of the (M0_PNDS_FAILED, M0_PNDS_OFFLINE, ...) states.
But now, the Motr client will switch to degraded mode on its own decision when I/O error happens.

I hope this will not lead to some other problems.